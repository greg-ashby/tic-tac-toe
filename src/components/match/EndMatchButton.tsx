type Props = {
  onEndMatchClick: () => void;
};

export function EndMatchButton({ onEndMatchClick }: Props) {
  const handleClick = () => {
    onEndMatchClick();
  };

  return (
    <button
      className="text-l font-bold bg-orange-500 text-white px-4 py-2 rounded hover:bg-orange-600 transition duration-200"
      type="button"
      onClick={handleClick}
    >
      End Match
    </button>
  );
}
