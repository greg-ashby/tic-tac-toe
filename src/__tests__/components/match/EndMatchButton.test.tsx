import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { EndMatchButton } from '@/components/match/EndMatchButton.tsx';

describe('EndMatchButton', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <EndMatchButton onEndMatchClick={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('calls onEndMatchClick when clicked', () => {
    const handleClick = jest.fn();
    const { getByRole } = render(
      <EndMatchButton onEndMatchClick={handleClick} />
    );

    const button = getByRole('button', { name: /end match/i });
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
