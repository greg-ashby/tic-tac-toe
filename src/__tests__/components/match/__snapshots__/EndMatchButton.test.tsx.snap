// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndMatchButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="text-l font-bold bg-orange-500 text-white px-4 py-2 rounded hover:bg-orange-600 transition duration-200"
    type="button"
  >
    End Match
  </button>
</DocumentFragment>
`;
